service/api/prettyprint.go

```diff
@@ -21,33 +21,48 @@ const (
 // SinglelineString returns a representation of v on a single line.
 func (v *Variable) SinglelineString() string {
 	var buf bytes.Buffer
-	v.writeTo(&buf, true, false, true, "", "")
+	v.writeTo(&buf, true, false, true, false, "", "")
+	return buf.String()
+}
+
+// SinglelineStringWithShortTypes returns a representation of v on a single line, with types shortened.
+func (v *Variable) SinglelineStringWithShortTypes() string {
+	var buf bytes.Buffer
+	v.writeTo(&buf, true, false, true, true, "", "")
 	return buf.String()
 }
 
 // SinglelineStringFormatted returns a representation of v on a single line, using the format specified by fmtstr.
 func (v *Variable) SinglelineStringFormatted(fmtstr string) string {
 	var buf bytes.Buffer
-	v.writeTo(&buf, true, false, true, "", fmtstr)
+	v.writeTo(&buf, true, false, true, false, "", fmtstr)
 	return buf.String()
 }
 
 // MultilineString returns a representation of v on multiple lines.
 func (v *Variable) MultilineString(indent, fmtstr string) string {
 	var buf bytes.Buffer
-	v.writeTo(&buf, true, true, true, indent, fmtstr)
+	v.writeTo(&buf, true, true, true, false, indent, fmtstr)
 	return buf.String()
 }
 
-func (v *Variable) writeTo(buf io.Writer, top, newlines, includeType bool, indent, fmtstr string) {
+func (v *Variable) typeStr(shortenType bool) string {
+	if shortenType {
+		return ShortenType(v.Type)
+	}
+
+	return v.Type
+}
+
+func (v *Variable) writeTo(buf io.Writer, top, newlines, includeType, shortenType bool, indent, fmtstr string) {
```

Comment:
I also considered turning the `includeType` bool into an enum with three values (off, short, full). That didn't work well though, because of the way includeType is negated sometimes.

Reply:
I think having four booleans on a row crosses a threshold. If @derekparker agrees I would rebase this on something like this https://github.com/go-delve/delve/pull/3536

Reply:
That would be fine with me. I suppose I'd wait until that PR is merged and then rebase on master? There's no rush, I'm now on vacation for two weeks, so can only continue with this after I'm back anyway.